### PR TITLE
MKL support for linux64

### PR DESCRIPTION
**Pull request by ericpre**
Add two variants: with/without with MKL support on linux64.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [n/a] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [n/a] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,8 +14,8 @@ source:
   sha256: 43616529f9b7d1afc83386f943dc66c4da5e052f00217ba7e3ad8dd1b5f3a825
 
 build:
-  number: 1
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  string: "{{ build_string }}py{{ py }}h{{ PKG_HASH }}_{{ build }}"
+  number: {{ build }}
```
Comment (isuruf) [inline recipe/meta.yaml:18]: To weigh down the mkl build
```suggestion
  number: {{ build }}
  track_features:
    - numexpr_mkl    # [linux64 and mkl_support == 'mkl']
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -20,13 +25,17 @@ requirements:
...
+    - mkl                                    # [linux64 and mkl_support == 'mkl']
+    - mkl-devel                              # [linux64 and mkl_support == 'mkl']
     - numpy
-  run:
+    - pip
     - python
     - setuptools
+  run:
+    - libblas * *mkl                         # [linux64 and mkl_support == 'mkl']
+    - mkl                                    # [linux64 and mkl_support == 'mkl']
```
Comment (isuruf) [inline recipe/meta.yaml:36]:
```suggestion
    - mkl                                    # [linux64 and mkl_support == 'mkl']
    - nomkl                                  # [linux64 and mkl_support != 'mkl']
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,4 +1,9 @@
 {% set version = "2.7.3" %}
+{% set build = 2 %}
+{% set build_string = '' %}
+
+{% set build_string = 'mkl_' %}              # [linux64 and mkl_support == 'mkl']
+{% set build = build + 100 %}                # [linux64 and mkl_support == 'mkl']
```
Comment (ericpre) [inline recipe/meta.yaml:6]: Should we removed changing the build number for `mkl` build? I can't find any documentation of how `track_feature` will interfere with build number - i.e which one will have priority. The closest documentation I have found on this topic is https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-channels.html?highlight=build%20number#managing-channels

Comment (isuruf) [reply recipe/meta.yaml:6]: Both are fine to have and make the mkl build have less priority, but in two different ways.

Comment (ericpre) [reply recipe/meta.yaml:6]: But here the mkl variant has higher build number, and I will expect that it get higher priority?

Comment (isuruf) [reply recipe/meta.yaml:6]: Oops. We should change it and make mkl variant has lower build number.

Comment (ericpre) [reply recipe/meta.yaml]: Done.

**Pull request merged**